File: C/greedy.c
#include <stdio.h>
#include <cs50.h>

int main (void)

{
    double dollar;
    int cent, score, nominalIndex;
    int coinsNominals[4] = { 25, 10, 5, 1 };
    do
    {
        printf("O hai! How mach change is owed?\n");
        dollar = GetDouble();
    }
    while (dollar < 0);
    //перевод долларов в центы
    cent = dollar*100;
    nominalIndex = 0;
    
    for(score = 0; cent > 0;)
    {
        if (cent >= coinsNominals[nominalIndex]) 
        {
            cent = cent - coinsNominals[nominalIndex];
            score++;
            continue;
        }
        nominalIndex++;
    }

    printf("%i\n", score);
}

File: Python/greedy.py
import cs50
coinsNominals = [25, 10, 5, 1]
index = 0
count = 0
g1 = 0
dollar = -1
while dollar < 0:
    print("O hai! How mach change is owed?")
    dollar = cs50.get_float()
cent = dollar * 100
while cent > 0:
    if cent >= coinsNominals[index]:
        cent -= coinsNominals[index]
        count += 1
        continue
    index += 1
print(count)